planner/mapper/mapper.go

```diff
@@ -119,36 +120,91 @@ func resolveOrderDependencies(
 
 	// If there is orderby, and any one of the condition fields that are join fields and have not been
 	// requested, we need to map them here.
+outer:
 	for _, condition := range source.Value().Conditions {
-		if len(condition.Fields) <= 1 {
-			continue
-		}
-
-		joinField := condition.Fields[0]
+		fields := condition.Fields[:] // copy slice
+		for {
+			numFields := len(fields)
+			if numFields == 2 {
```

Comment:
If you did this you could then move the contents of `resolveChildOrder` into this function and remove the pointer based mutation stuff that I was going to otherwise comment about (param `existingFields`), as `resolveChildOrder` will only be called once.

Reply:
I wanted each `if/elseif/else` to be explicit, as they need to handle different cases, so I didnt want to get too smart about shared logic between them.

I'd also prefer to keep `resolveChildOrder` to not make this function more complex then it needs to be as its already a double `for` loop and a muli-level `else if`. Strong preference to keep it out.

As for the pointer based mutation stuff on the `existingFields` that was like this before this PR, so im just keeping the existing semantics.

lmk

Reply:
>  I didnt want to get too smart about shared logic between them.

I dont think what I'm suggesting is particularly fancy, it is the same code, just with reduced duplication and if-else scope.  I dont see why you wouldnt want that.

> Strong preference to keep it out.

Fair enough, I will review it then.

Reply:
> Fair enough, I will review it then.

Looked it over - I hate the input pointer mutation stuff, but that is inherited from this function itself (I missed that yesterday), and whilst if that variable was not a pointer in `resolveOrderDependencies` it would be easy enough to write `resolveChildOrder` in such a way to avoid the pointer stuff, it is not, and changing that is much more of a pain and out of scope.

I do also dislike that `resolveChildOrder` is really two functions pretending to be one, on account of the if-else, but that is fairly minor and easy to change in the future if anyone feels the need to.